<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Selection</class>
 <widget class="QDockWidget" name="Selection">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>613</width>
    <height>409</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Selection</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <layout class="QVBoxLayout" name="verticalLayout_6">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QPushButton" name="Create_selection_item_button">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Create Selection</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="Clear_all_button">
        <property name="toolTip">
         <string extracomment="Clear selection for All Types"/>
        </property>
        <property name="text">
         <string>Clear Selection</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QCheckBox" name="hl_checkBox">
        <property name="text">
         <string>Highlight</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTabWidget" name="selectionOrEuler">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Selection</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QTabWidget" name="tabWidget">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tab">
             <attribute name="title">
              <string>Simplices</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="1,1">
                <item>
                 <widget class="QLabel" name="label_2">
                  <property name="text">
                   <string>Selection Type :</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QComboBox" name="Selection_type_combo_box">
                  <item>
                   <property name="text">
                    <string>Vertex</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Facet</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Edge</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Connected Component (Facet)</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Path Between Vertices</string>
                   </property>
                  </item>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer_5">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_11" stretch="0,1,0,1">
                <item>
                 <widget class="QCheckBox" name="lassoCheckBox">
                  <property name="text">
                   <string>Lasso</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_3">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <layout class="QVBoxLayout" name="verticalLayout_7">
                  <item>
                   <widget class="QPushButton" name="Select_all_NTButton">
                    <property name="text">
                     <string>Select All Non Triangle</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="Select_boundaryButton">
                    <property name="text">
                     <string>Select Boundary Edges</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <widget class="QPushButton" name="Add_to_selection_button">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>Add to Selection</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout">
                <item>
                 <widget class="QRadioButton" name="Insertion_radio_button">
                  <property name="text">
                   <string>Insertion</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QRadioButton" name="Removal_radio_button">
                  <property name="text">
                   <string>Removal</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>Brush &amp;size:</string>
                  </property>
                  <property name="buddy">
                   <cstring>Brush_size_spin_box</cstring>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="Brush_size_spin_box"/>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_2">
             <attribute name="title">
              <string>Components</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_5">
                <item>
                 <widget class="QLabel" name="label_3">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>Threshold:</string>
                  </property>
                  <property name="buddy">
                   <cstring>Threshold_size_spin_box</cstring>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="Threshold_size_spin_box">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="maximum">
                   <number>999999999</number>
                  </property>
                  <property name="value">
                   <number>8</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="Get_minimum_button">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>&amp;Get Minimum</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer_7">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="Select_isolated_components_button">
                <property name="text">
                 <string>Select &amp;Isolated Components Below Threshold</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer_6">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <item>
                 <widget class="QLabel" name="label_6">
                  <property name="text">
                   <string>Sharp edges angle:</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="Sharp_angle_spinbox">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="maximum">
                   <number>180</number>
                  </property>
                  <property name="value">
                   <number>60</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="Select_sharp_edges_button">
                  <property name="text">
                   <string>Select</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <spacer name="verticalSpacer_10">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QPushButton" name="Expand_reduce_button">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Expand/Reduce</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="Expand_reduce_spin_box">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimum">
                 <number>-50</number>
                </property>
                <property name="maximum">
                 <number>50</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="Regularize_button">
              <property name="text">
               <string>Regularize</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Clear_button">
              <property name="toolTip">
               <string extracomment="Clear selection for Selection Type"/>
              </property>
              <property name="text">
               <string>&amp;Clear Current Type</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Inverse_selection_button">
              <property name="toolTip">
               <string extracomment="Invert selection for Selection Type"/>
              </property>
              <property name="text">
               <string>Invert Selection</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Select_all_button">
              <property name="toolTip">
               <string extracomment="Select all simplices of Selection Type"/>
              </property>
              <property name="text">
               <string>Select &amp;All</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_8">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_13" stretch="1,0">
          <item>
           <widget class="QComboBox" name="operationsBox">
            <item>
             <property name="text">
              <string>Create Point Set from Selected Vertices</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Create Polyline from Selected Edges</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Create Facegraph from Selected Facets</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Erase Selected Facets</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Keep Connected Components of Selected Facets</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Expand Face Selection to Stay Manifold After Removal</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Convert from Edge Selection to Facets Selection</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Convert from Edge Selection to Point Selection</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Convert from Facet Selection to Boundary Edge Selection</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Convert from Facet Selection to Point Selection</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="validateButton">
            <property name="text">
             <string>Validate</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Edition</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QGroupBox" name="edition_groupBox">
          <property name="title">
           <string>Euler Operations</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <spacer name="verticalSpacer_9">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_9" stretch="0,1,0,0">
             <item>
              <widget class="QComboBox" name="editionBox">
               <item>
                <property name="text">
                 <string>Join vertex</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Split vertex</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Split edge</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Join face</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Split face</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Collapse edge</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Flip edge</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Add center vertex</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Remove center vertex</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Add vertex and face to border (Advanced)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Add face to border (Advanced)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Move point</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="instructionsLabel">
               <property name="text">
                <string>Instructions 

</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="docImage_Label">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Ctrl+Z to cancel the temporary selection. Ctrl+U to undo last operation (if applicable).</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
